src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ManagedProjectSystemPackage.cs

```diff
@@ -33,12 +32,8 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
         {
             IComponentModel componentModel = await this.GetServiceAsync<SComponentModel, IComponentModel>();
             
-            JoinableTaskContext context = componentModel.GetService<JoinableTaskContext>();
-
             IEnumerable<IPackageService> packageServices = componentModel.GetExtensions<IPackageService>();
 
-            await context.Factory.SwitchToMainThreadAsync(cancellationToken);
-
             foreach (IPackageService packageService in packageServices)
             {
                 await packageService.InitializeAsync(this);
```

Comment:
Do we actually need to wait for all of these, as opposed to fire-and-forgetting them? Presumably the order that MEF finds them is indeterminant, so having dependencies between IPackageServices would be a bad design anyway.

Reply:
Yes we do, the project selector and project factory _**must be**_ registered by the time the package has finished initializing; the solution has initialized the package synchronously on the UI thread when opening a project so that they register them, if they are not registered by the time the package has finished initializing, the projects (F#, XProj) will fail to load correctly.
There will be very little value kicking off the others asynchronously anyway - they do very little.

Reply:
I don't think it'd be incorrect to `await WhenAll` them. I'll try it out.